scripts/create_role_molecule.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import yaml
+
+PROJECT_DIR = Path(__file__).resolve().parent.parent
+SCRIPT_DIR = PROJECT_DIR / 'scripts'
+CI_TEMPLATES_DIR = PROJECT_DIR / 'ci' / 'templates'
+CI_CONFIG_DIR = PROJECT_DIR / 'ci' / 'config'
+ROLES_DIR = PROJECT_DIR / 'ci_framework' / 'roles'
+ZUUL_JOB_DIR = PROJECT_DIR / 'zuul.d'
+SOURCE_ROLES_DOCS_DIR = PROJECT_DIR / 'docs' / 'source' / 'roles'
+
+
+def generate_zuul_job_role_yaml(role_name):
+    """
+    Generate YAML structure for a Zuul job based on given role.
+
+    Args:
+        role_name (str): Name of the role.
+
+    Returns:
+        dict: YAML structure for the role.
+    """
+    zuul_job_base_yaml = {
+        'job': {
+            'name': f'cifmw-molecule-{role_name}',
+            'parent': 'cifmw-molecule-base',
+            'vars': {
+                'TEST_RUN': role_name
+            },
+            'files': [
+                '^ansible-requirements.txt',
+                '^molecule-requirements.txt',
+                f'^ci_framework/roles/{role_name}/(?!meta|README).*',
+                '^ci/playbooks/molecule.*'
+            ]
+        }
+    }
+    return zuul_job_base_yaml
+
+
+def regenerate_projects_zuul_jobs_yaml():
+    """
+    Regenerate zuul.d/projects.yaml file with the name of each role
+    under ci-framework/roles
+    """
+    project_yaml_path = CI_TEMPLATES_DIR / 'projects.yaml'
+
+    try:
+        with open(project_yaml_path, 'r') as file:
+            projects_jobs_info = yaml.safe_load(file)
+    except FileNotFoundError as e:
+        raise e
+
+    # Add each role name after the generated content
+    for role_directory in ROLES_DIR.iterdir():
+        if not role_directory.is_dir():
+            print("Skipping. Not a role directory")
+            continue
+        projects_jobs_info[0]['project']['github-check']['jobs'].append(
+            role_directory.name
+        )
+
+    try:
+        with open(ZUUL_JOB_DIR / 'projects.yaml', 'w') as projects_file:
+            yaml.dump(projects_jobs_info, projects_file)
+    except Exception as e:
+        raise e
+
+
+def regenerate_molecule_zuul_jobs_yaml():
+    """
+    Regenerate zuul.d/molecule.yaml file with role-specific YAML structures.
+    """
+    molecule_yaml_zuul_jobs = []
+
+    for role_directory in ROLES_DIR.iterdir():
+        if not role_directory.is_dir():
+            print("Skipping. Not a role directory")
+            continue
+
+        try:
+            role_yaml = generate_zuul_job_role_yaml(role_directory.name)
+            molecule_yaml_zuul_jobs.append(role_yaml)
+        except Exception as e:
```

Comment:
Why is this needed? Re raising it here means nothing...

Reply:
True. Removed. Thx.